libs/repositories/src/UsdRepository/UsdRepository.ts

```diff
@@ -35,17 +35,40 @@ export interface UsdRepository {
 }
 
 export class UsdRepositoryNoop implements UsdRepository {
-  getUsdPrice(
+  async getUsdPrice(
     chainId: SupportedChainId,
     tokenAddress: string
-  ): Promise<number> {
+  ): Promise<number | null> {
     return null;
   }
-  getUsdPrices(
+  async getUsdPrices(
     chainId: SupportedChainId,
     tokenAddress: string,
     priceStrategy: PriceStrategy
   ): Promise<PricePoint[] | null> {
     return null;
   }
 }
+
+export const serializePricePoints = (pricePoints: PricePoint[]): string => {
+  console.log('pricePoints', pricePoints);
+  const serialized = pricePoints.map((point) => ({
+    ...point,
+    date: point.date.toISOString(),
+  }));
```

Comment:
Nitpick, feel free to ignore: use a custom serializer/deserializer instead of "manually" serializing some fields before/after JSON stringify.

Also, JSON already does that when serializing dates:

![image](https://github.com/user-attachments/assets/77271d34-370d-4e8e-8ab7-59d2e22ca272)

You will still need to recreate the date when parsing, however.

Reply:
Good point, I will make an issue. I briefly check and I like this library https://github.com/cowprotocol/bff/issues/64